### PR TITLE
IA-2760: Update default Projects (mobile app IDs)

### PR DESCRIPTION
**Pull request by butofleury**
Related JIRA tickets : [IA-2760](https://bluesquare.atlassian.net/browse/IA-2760)

## Self proofreading checklist

- [x] Did I use eslint and black formatters
- [ ] Is my code clear enough and well documented
- [ ] Are my typescript files well typed
- [ ] New translations have been added or updated if new strings have been introduced in the frontend
- [ ] My migrations file are included
- [ ] Are there enough tests
- [ ] Documentation has been included (for new feature)

## Changes

Created 3 additional projects: **Planning, Georegistry/Géoregistre and Children vaccination/Vaccination des enfants** by adding an argument in the script launcher to specify the creation of 3 additionnals projects!

## How to test

From setuper folder, launch setuper via `python3 setuper.py --additionnal_projects` to generate test data. 
Then login with the created account and open projects page: [http://localhost:8081/dashboard/settings/projects/](http://localhost:8081/dashboard/settings/projects/).

You should see 3 additional projects: **Planning, Georegistry/Géoregistre and Children vaccination/Vaccination des enfants**

From the forms page, check for the forms: 
  - Equipment/Pop/Social mob./Microplans: you should see 2 additionals projects(Planning and Georegistry/Géoregistre)
  - Child - Registration and Child - Follow up: 1 additional project(Children vaccination/Vaccination des enfants)

## Print screen / video

https://github.com/BLSQ/iaso/assets/6383219/52c44ffc-b835-410d-a3d8-bc0e428f7101

[IA-2760]: https://bluesquare.atlassian.net/browse/IA-2760?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: setuper/additional_projects.py [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+    if len(data_source) > 0:
+        default_project = data_source[0]["projects"][0][0]["id"]
+        project_ids.append(default_project)
+        data_source[0]["project_ids"] = project_ids
+        data_source[0]["default_version_id"] = data_source[0]["default_version"]["id"]
+        iaso_client.put(f"/api/datasources/{data_source[0]['id']}/", json=data_source[0])
+
+
+def create_projects(account_name, iaso_client):
+    print(f"-- Creating 3 additional projects for account :{account_name}")
```
Comment (kemar) [inline setuper/additional_projects.py]:
```suggestion
    print(f"-- Creating 3 additional projects for account: {account_name}")
```

File: setuper/additional_projects.py [hunk 2]
```diff
@@ -0,0 +1,117 @@
+def projects_mapper(account_name):
+    projects = [
+        {
+            "name": "Planning",
+            "app_id": f"{account_name}.planning",
+            "feature_flags": ["REQUIRE_AUTHENTICATION", "TAKE_GPS_ON_FORM", "", "FORMS_AUTO_UPLOAD", "PLANNING"],
```
Comment (kemar) [inline setuper/additional_projects.py]:
```suggestion
            "feature_flags": ["REQUIRE_AUTHENTICATION", "TAKE_GPS_ON_FORM", "FORMS_AUTO_UPLOAD", "PLANNING"],
```

I think that the empty perm is a typo?

File: setuper/additional_projects.py [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+        iaso_client.put(f"/api/datasources/{data_source[0]['id']}/", json=data_source[0])
+
+
+def create_projects(account_name, iaso_client):
+    print(f"-- Creating 3 additional projects for account :{account_name}")
+    projects = projects_mapper(account_name)
+    created_projects = []
+    for project in projects:
+        project["linked_forms"] = None
+        project["feature_flags"] = feature_flags_mapper(project, iaso_client)
```
Comment (kemar) [inline setuper/additional_projects.py]:
```suggestion

    flags = iaso_client.get("/api/featureflags/")
    existing_flags = set([flag["code"] for flag in flags["featureflags"]])

    for project in projects:
        project["linked_forms"] = None
        project["feature_flags"] = set(project["feature_flags"]).intersection(existing_flags)
```

We could save some HTTP queries by avoiding calling `feature_flags_mapper() -> GET /api/featureflags/` in each iteration of the loop.

What do you think?

**Review by kemar — CHANGES_REQUESTED** [review]

File: setuper/additional_projects.py (hunk 2 above)
Comment (butofleury) [reply setuper/additional_projects.py]: yes, It was a mistake from my side! 
Thanks!

File: setuper/additional_projects.py (hunk 3 above)
Comment (butofleury) [reply setuper/additional_projects.py]: Nice idea. But just a small change as the  payload for `post: /api/apps/` need the whole object for project features flags , I would suggest:
-  to remove line 78 and keep all existing flags 
- on line 82, to filter on existing flags to keep only flags defined on the current project : `project["feature_flags"] = [flag for flag in flags["featureflags"] if flag["code"] in project["feature_flags"]]`


The **create_projects** would look like:

```
def create_projects(account_name, iaso_client):
    print(f"-- Creating 3 additional projects for account: {account_name}")
    projects = projects_mapper(account_name)
    created_projects = []
    flags = iaso_client.get("/api/featureflags/")

    for project in projects:
        project["linked_forms"] = None
        project["feature_flags"] = [flag for flag in flags["featureflags"] if flag["code"] in project["feature_flags"]]
        new_project = iaso_client.post("/api/apps/", json=project)
        created_projects.append(new_project)

    project_ids = get_project_ids(created_projects, iaso_client)
    update_org_unit_types_with_new_projects(iaso_client, project_ids)
    link_created_projects_to_main_data_source(account_name, iaso_client, project_ids)
```

Comment (kemar) [reply setuper/additional_projects.py]: Yup, this works!

**Review by kemar — APPROVED** [review]
LGTM

**Pull request merged**